Src/Web/Web.Shared.Net/RequestTrackingTelemetryModule.cs

```diff
@@ -230,12 +230,23 @@ internal bool NeedProcessRequest(HttpContext httpContext)
                 return false;
             }
 
-            if (httpContext.Response.StatusCode < 400)
+            if (httpContext.Response.StatusCode < 400 && httpContext.Handler != null)
             {
-                if (this.IsHandlerToFilter(httpContext.Handler))
+                string handlerName = httpContext.Handler.GetType().FullName;
+                if (this.IsHandlerToFilter(handlerName))
                 {
                     return false;
                 }
+
+                string requestPath = httpContext.Request.UnvalidatedGetPath();
+                if (string.Equals(handlerName, RequestTrackingConstants.TransferHandlerType, StringComparison.Ordinal)
```

Comment:
These comparisons seem to happen for every successful request; can we think of some tweaks for performance of those three comparisons?
- "System.Web.Handlers" seems to be a common part for many handlers (but at the same time protects us from disabling similarly named handler in the other namespace), will using Hash comparison be faster (maybe not...)?
- We seem to use full path of the request for indexOf(), is there any parsed request path object available where we can only use part of the path for indexOf()?